File: score-http/score-http/src/main/java/org/oagi/score/gateway/http/api/library_management/data/Library.java
package org.oagi.score.gateway.http.api.library_management.data;

import lombok.Data;
import org.oagi.score.repo.api.user.model.ScoreUser;

import java.math.BigInteger;
import java.util.Date;

@Data
public class Library {

    private BigInteger libraryId;

    private String name;

    private String organization;

    private String link;

    private String domain;

    private String description;

    private String state;

    private ScoreUser createdBy;

    private ScoreUser lastUpdatedBy;

    private Date creationTimestamp;

    private Date lastUpdateTimestamp;

}

File: connect-center-api/app/repositories/models/library.py
"""Repository row model for library resources."""


from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class LibraryRow(BaseModel):
    """Represent LibraryRow."""
    library_id: int
    name: str
    type: str | None = None
    organization: str | None = None
    description: str | None = None
    link: str | None = None
    domain: str | None = None
    state: str | None = None
    is_read_only: bool
    is_default: bool
    created_by: int
    creation_timestamp: datetime
    last_updated_by: int
    last_update_timestamp: datetime

    model_config = ConfigDict(frozen=True, from_attributes=True)


class LibrarySummaryRow(BaseModel):
    """Repository summary row for library references."""

    library_id: int
    name: str

    model_config = ConfigDict(frozen=True, from_attributes=True)
